airflow/providers/apache/kafka/operators/consume.py

```diff
@@ -0,0 +1,189 @@
+        super().__init__(**kwargs)
+
+        self.topics = topics
+        self.apply_function = apply_function
+        self.apply_function_batch = apply_function_batch
+        self.apply_function_args = apply_function_args or ()
+        self.apply_function_kwargs = apply_function_kwargs or {}
+        self.kafka_config_id = kafka_config_id
+        self.commit_cadence = commit_cadence
+        self.max_messages = max_messages or True
+        self.max_batch_size = max_batch_size
+        self.poll_timeout = poll_timeout
+
+        if self.max_messages is True:
+            self.READ_TO_END = True
+        else:
+            self.READ_TO_END = False
+
+        if self.commit_cadence not in VALID_COMMIT_CADENCE:
+            raise AirflowException(
+                f"commit_cadence must be one of {VALID_COMMIT_CADENCE}. Got {self.commit_cadence}"
+            )
+
+        if self.max_messages and self.max_batch_size > self.max_messages:
+            self.log.warning(
+                "max_batch_size (%s) > max_messages (%s). Setting max_messages to %s ",
+                self.max_batch_size,
+                self.max_messages,
+                self.max_batch_size,
+            )
+
+        if self.commit_cadence == "never":
+            self.commit_cadence = None
+
+        if apply_function and apply_function_batch:
+            raise AirflowException(
+                "One of apply_function or apply_function_batch must be supplied, not both."
+            )
+
+    def execute(self, context) -> Any:
+
+        consumer = KafkaConsumerHook(topics=self.topics, kafka_config_id=self.kafka_config_id).get_consumer()
+
+        if isinstance(self.apply_function, str):
+            self.apply_function = import_string(self.apply_function)
+
+        if isinstance(self.apply_function_batch, str):
+            self.apply_function_batch = import_string(self.apply_function_batch)
+
+        if self.apply_function:
+            apply_callable = partial(
+                self.apply_function, *self.apply_function_args, **self.apply_function_kwargs  # type: ignore
+            )
+
+        if self.apply_function_batch:
+            apply_callable = partial(
+                self.apply_function_batch,  # type: ignore
+                *self.apply_function_args,
+                **self.apply_function_kwargs,
+            )
+
+        messages_left = self.max_messages
+        messages_processed = 0
+
+        while self.READ_TO_END or (
+            messages_left > 0
+        ):  # bool(True > 0) == True in the case where self.max_messages isn't set by the user
+
+            if not isinstance(messages_left, bool):
+                batch_size = self.max_batch_size if messages_left > self.max_batch_size else messages_left
+            else:
+                batch_size = self.max_batch_size
+
+            msgs = consumer.consume(num_messages=batch_size, timeout=self.poll_timeout)
+            messages_left -= len(msgs)
+            messages_processed += len(msgs)
```

Comment:
I think `messages_processed` is not used too

Reply:
good catch, dead code from earlier versions